### PR TITLE
Feat/llamaindex-ts

### PR DESCRIPTION
**Pull request by abhishekpatil4**
- Added support for LlamaIndex TS framework in JS SDK
- Created test cases for framework validation

### CODE REVIEW

File: js/src/frameworks/llamaindex.spec.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import { beforeAll, describe, expect, it } from "@jest/globals";
+import { z } from "zod";
+import { getTestConfig } from "../../config/getTestConfig";
+import { ActionExecuteResponse } from "../sdk/models/actions";
+import { LlamaIndexToolSet } from "./llamaindex";
+
+describe("Apps class tests", () => {
```
Comment (shreysingla11) [inline js/src/frameworks/llamaindex.spec.ts:7]: The test suite description `Apps class tests` is misleading as it's testing the LlamaIndexToolSet class. Consider renaming it to `LlamaIndexToolSet tests` for better clarity.

File: js/src/frameworks/llamaindex.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  ) {
+    super({
+      apiKey: config.apiKey || null,
+      baseUrl: config.baseUrl || COMPOSIO_BASE_URL,
+      runtime: config?.runtime || null,
+      entityId: config.entityId || LlamaIndexToolSet.DEFAULT_ENTITY_ID,
+    });
+  }
+
+  private _wrapTool(
```
Comment (shreysingla11) [inline js/src/frameworks/llamaindex.ts:35]: The `_wrapTool` method accepts `schema: any` as a parameter type. Consider creating a proper interface for better type safety:

```typescript
interface ToolSchema {
  name: string;
  description: string;
  parameters: {
    properties?: Record<string, unknown>;
    required?: string[];
  };
}
```

File: js/src/frameworks/llamaindex.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    schema: any,
+    entityId: Optional<string> = null
+  ): FunctionTool<Record<string, unknown>, JSONValue | Promise<JSONValue>> {
+    return FunctionTool.from(
+      async (params: Record<string, unknown>) => {
```
Comment (shreysingla11) [inline js/src/frameworks/llamaindex.ts:40]: Consider adding error handling for the JSON parsing operation:

```typescript
try {
  return JSON.parse(JSON.stringify(result)) as JSONValue;
} catch (error) {
  throw new Error(`Failed to parse tool result: ${error.message}`);
}
```

This would provide better error messages and make debugging easier.

**Comment by shreysingla11** [comment]
### Code Review Summary

#### Overall Assessment
The implementation of LlamaIndex TypeScript framework support is well-structured and follows the established patterns in the codebase. The code is generally clean and functional, with good test coverage.

#### Positive Aspects
- Good integration with the base `ComposioToolSet` class
- Proper implementation of LlamaIndex's `FunctionTool` interface
- Clean removal of unused OpenAI-specific code
- Good test coverage for basic functionality

#### Suggestions for Improvement
1. **Type Safety**
   - Replace `any` type in `_wrapTool` with a proper interface
   - Add proper error handling for JSON operations

2. **Testing**
   - Rename test suite to accurately reflect what's being tested
   - Consider adding error case tests
   - Add tests for edge cases with empty or invalid filters

3. **Documentation**
   - Add JSDoc comments for public methods
   - Consider adding examples in comments for common use cases

#### Code Quality Rating: 8/10
The code is well-implemented and follows good practices, with minor improvements suggested for type safety and documentation.

**Comment by plxity** [comment]
@abhishekpatil4 Couple of actions are failing, please fix them

**Review by himanshu-dixit — APPROVED** [review]

**Comment by himanshu-dixit** [comment]
@abhishekpatil4 https://github.com/ComposioHQ/composio/blob/d60252281a5705dbd7ccabe4bc51545ddf14a5a1/js/src/frameworks/langgraph.ts

Can follow implementation like this
